tests/optimize/test_optimize.py

```diff
@@ -115,17 +115,21 @@ def test_nontrainable_data(self, opt):
         """Check non-trainable argument does not get updated"""
 
         def func(a, b, c, d):
+            if not qml.math.allclose(b, 1.0):
+                # Tests Nesterov momentum optimizer only shifts arguments of the trainable parameters
+                raise ValueError("Only accepts a b of 1.0.")
             return a * b * c * d
 
-        a = np.array(1.0, requires_grad=True)
+        a = np.array(0.1, requires_grad=True)
```

Comment:
@albi3ro it seems that the change in this test doesn't raise an error similar to the one in the demo. Is that correct? What could be a good test case to catch the edge case from the demo?

Reply:
For me, this test failed on the release candidate branch.

Reply:
That's the trick, thanks for pointing it out!

Any reasons for going with an error instead of an `assert`?